src/main/java/team/gdsc/code2cv/feature/auth/service/AuthService.java

```diff
@@ -0,0 +1,110 @@
+package team.gdsc.code2cv.feature.auth.service;
+
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import team.gdsc.code2cv.feature.auth.dto.AuthReq;
+import team.gdsc.code2cv.feature.auth.dto.AuthRes;
+import team.gdsc.code2cv.feature.auth.repository.GithubClient;
+import team.gdsc.code2cv.feature.user.domain.GithubAccount;
+import team.gdsc.code2cv.feature.user.domain.UserCommand;
+import team.gdsc.code2cv.feature.user.entity.User;
+import team.gdsc.code2cv.feature.user.repository.UserRepository;
+import team.gdsc.code2cv.global.jwt.JwtProvider;
+import team.gdsc.code2cv.global.jwt.JwtToken;
+import team.gdsc.code2cv.global.jwt.JwtUser;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class AuthService {
+	private final JwtProvider jwtProvider;
+	private final UserRepository userRepository;
+	private final BCryptPasswordEncoder bCryptPasswordEncoder;
+	private final GithubClient githubClient;
+
+	/**
+	 * 깃허브 로그인 또는 회원가입
+	 * 1. 깃허브로부터 받은 code와 state를 이용하여 깃허브 access token을 받아온다.
+	 * 2. 깃허브 access token을 이용하여 깃허브 계정 정보를 받아온다.
+	 * 3. 깃허브 계정 정보를 이용하여 회원가입 또는 로그인을 진행한다.
+	 * 4. 로그인시, 내정보가 변경되었을 경우, 이를 반영한다.
+	 */
+	public AuthRes.LoginResponse githubLoginOrSignUp(AuthReq.GithubLoginRequest req) {
+		String githubAccessToken = githubClient.getGithubAccessToken(req.code(), req.state());
+
+		GithubAccount githubAccount = githubClient.getGithubAccountResponse(githubAccessToken);
+
+		User user = userRepository.findByGithubAccountGithubId(githubAccount.getGithubId())
+			.orElseGet(() -> signUp(githubAccount));
+
+		user.updateGithubAccount(githubAccount);
+
+		JwtToken jwtToken = createToken(user);
+		return AuthRes.LoginResponse.from(jwtToken, user);
+	}
+
+	/**
+	 * 액세스 토큰 재발급
+	 */
+	public AuthRes.AccessTokenResponse refreshAccessToken(String rawToken) {
```

Comment:
```suggestion
	public AuthRes.AccessTokenResponse reissueAccessToken(String rawToken) {
```

1. refreshToken과 헷갈릴 가능성 존재
2. jwtProvider.reissueAccessToken 메소드 존재
의 이유로 이렇게 수정하는 건 어떨까요?

Reply:
인정합니다